Code/docker.js

```diff
@@ -1710,29 +1710,39 @@ define([
             // on keyup for .wcFrameButtonBar > .wcFrameButton
             function __onKeyUpPanelButton(e) {
                 if(e.which === 13 || e.which === 32) {
+                    // wcDisableSelection will disable text selection, remove it before performing respective action
                     $('body').removeClass('wcDisableSelection');
+                    // Get frame instance on which event is triggered and perform respective action
                     for (var i = 0; i < self._frameList.length; ++i) {
                         var frame = self._frameList[i];
+                        // check if event target is in frame container
                         if(frame.$frame.find(e.target).length > 0) {
                             if(frame.$close[0] == e.target) {
                                 self.__closePanel(frame.panel());
+                                e.stopPropagation();
                                 return;
                             } else if(frame.$tabLeft[0] == e.target) {
                                 frame._tabScrollPos -= frame.$tabBar.width() / 2;
                                 if (frame._tabScrollPos < 0) {
                                     frame._tabScrollPos = 0;
                                 }
-                                frame.__updateTabs();
+                                // perform tab scroll in async mode and set focus back to scroll left button
+                                setTimeout(function() {
+                                    frame.__updateTabs();
+                                    e.target.focus();
```

Comment:
Save the e.target in temporary variable and use it to do the operation as it is happening in the asynchronous mode, and we won't know - how browser has implemented the event management.

It can reuse the same event object for garbage collection.

Fix similar issue in the below implementation as well.

Also - for consistency use capital letter for syntax start (wherever applicable).